### PR TITLE
Page title is not correctly updated (#296)

### PR DESCRIPTION
**Pull request by sumiet**
Hey, I am new to OSS and this would be my first contribution. Please review the PR.

### CODE REVIEW

**Comment by sumiet** [comment]
I am a little bit confused here, the build is working perfectly for my local and also I haven't modified anything on `package.json`. Why is ```npm install``` still failing. Please guide, I am new to the community and not much aware about the procedure here.

**Comment by willshowell** [comment]
Thanks for the PR! 

The Travis builds have been failing for a while due to some other commit.

I think this behavior should be added to [`ComponentCategoryList `](https://github.com/angular/material.angular.io/blob/master/src/app/pages/component-category-list/component-category-list.ts), instead of `ComponentSidenav`. Also, any subscription needs to be unsubscribed from in `ngOnDestroy`.

**Comment by sumiet** [comment]
@willshowell I have implemented the comments, please review.

File: src/app/pages/component-category-list/component-category-list.ts [hunk 1]
```diff
@@ -26,6 +28,16 @@ export class ComponentCategoryList implements OnInit {
     this.params = Observable.combineLatest(
       this._route.pathFromRoot.map(route => route.params),
       Object.assign);
+
+    // title on topbar navigation
+    this.routeParamSubscription = this._route.parent.parent.params.subscribe( params => {
+      let sectionName = params['section'];
+      this._componentPageTitle.title = SECTIONS[sectionName];
+    });
```
Comment (willshowell) [inline src/app/pages/component-category-list/component-category-list.ts]: Try this, it's a little simpler

```ts
this.routeParamSubscription = this.params.subscribe(params => {
  const sectionName = params['section'];
  this._componentPageTitle.title = SECTIONS[sectionName];;
});
```

Comment (sumiet) [reply src/app/pages/component-category-list/component-category-list.ts]: Thanks @willshowell, implemented the changes. Please review.

**Comment by willshowell** [comment]
Pinging @jelbourn for review

**Comment by sumiet** [comment]
@jelbourn can you please review this.

**Comment by sumiet** [comment]
@willshowell shall I wait for the reviews? or should I try something else. I am really new to this please guide.

**Comment by willshowell** [comment]
@sumiet I think your approach is fine - the team will let you know if you need to try something else. I can't speak for them, but it looks like there is a race to get a stable release out soon, so fixing docs site bugs are probably a lower priority than usual.

**Comment by jelbourn** [comment]
I'm trying to make the CI green first presently

**Comment by sumiet** [comment]
@willshowell @jelbourn Okay thanks for the updates.

**Comment by jelbourn** [comment]
@amcdnl can you review?

**Pull request merged**